### ISSUE TITLE
Document the precedence /behavior in case of multiple package configuration providers

### ISSUE THREAD

**Issue by sschuberth** [oss-review-toolkit/ort#8742]
This

https://github.com/oss-review-toolkit/ort/blob/811daef3d35abb440a1ad044bcbf14738fc95767/model/src/main/resources/reference.yml#L42-L51

should be documented similar to curation providers at

https://github.com/oss-review-toolkit/ort/blob/811daef3d35abb440a1ad044bcbf14738fc95767/model/src/main/resources/reference.yml#L53-L55

**Comment by sschuberth** [comment]
It probably makes sense to implement https://github.com/oss-review-toolkit/ort/issues/6972 first.

**Comment by mnonnenmacher** [comment]
> It probably makes sense to implement [#6972](https://github.com/oss-review-toolkit/ort/issues/6972) first.

Multiple package configurations from different sources should already be allowed since #5334.

**Comment by sschuberth** [comment]
> Multiple package configurations from different sources should already be allowed since [#5334](https://github.com/oss-review-toolkit/ort/pull/5334).

Right, but not if they apply to the same ID / provenance. The mentioned [check is still there](https://github.com/oss-review-toolkit/ort/blob/89902b9f079ef13048a24d3435a32a3b1196de34/plugins/package-configuration-providers/api/src/main/kotlin/SimplePackageConfigurationProvider.kt#L65-L68).

**Comment by mnonnenmacher** [comment]
> > Multiple package configurations from different sources should already be allowed since [#5334](https://github.com/oss-review-toolkit/ort/pull/5334).
> 
> Right, but not if they apply to the same ID / provenance. The mentioned [check is still there](https://github.com/oss-review-toolkit/ort/blob/89902b9f079ef13048a24d3435a32a3b1196de34/plugins/package-configuration-providers/api/src/main/kotlin/SimplePackageConfigurationProvider.kt#L65-L68).

That check is only internal in the `SimplePackageConfigurationProvider` and not applied to package configurations from different providers, so it's not really related to this issue which is about the behavior with multiple providers. See the commit message of 00a84870cb683bc02dc15325cd43daf18deff038.
